tests/otel_tracing_e2e/test_e2e.py

```diff
@@ -0,0 +1,83 @@
+import json
+import os
+
+from opentelemetry import trace, propagate
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+from opentelemetry.sdk.resources import DEPLOYMENT_ENVIRONMENT, Resource, SERVICE_NAME
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+from opentelemetry.semconv.trace import SpanAttributes
+from opentelemetry.propagators.b3 import B3MultiFormat, B3SingleFormat
+from _validator import validate_trace
+from utils import context, weblog, interfaces, scenarios, missing_feature
+
+
+class E2ETestBase:
+    def setup_main(self):
+        self.setup_opentelemetry()
+        self.use_128_bits_trace_id = False
+        with self.tracer.start_as_current_span(name="runner.get", kind=trace.SpanKind.CLIENT) as span:
+            span.set_attribute(SpanAttributes.HTTP_METHOD, "GET")
+            span.set_attribute(SpanAttributes.HTTP_URL, "http://weblog:7777")
+            headers = {}
+            propagate.get_global_textmap().inject(headers)
+            self.r = weblog.get(path="/", headers=headers)
+            span.set_attribute(SpanAttributes.HTTP_STATUS_CODE, self.r.status_code)
+            self.trace_id = span.get_span_context().trace_id
+            self.root_span_id = span.get_span_context().span_id
+
+    def setup_opentelemetry(self):
+        resource = Resource.create(
+            attributes={SERVICE_NAME: "system-tests-runner", DEPLOYMENT_ENVIRONMENT: "system-tests"}
+        )
+        dd_site = os.environ.get("DD_SITE")
+        if dd_site is None or dd_site == "":
+            dd_site = "datad0g.com"
+        exporter = OTLPSpanExporter(
+            endpoint=f"https://trace.agent.{dd_site}/api/v0.2/traces",
+            headers={"dd-protocol": "otlp", "dd-api-key": os.environ.get("DD_API_KEY"),},
+        )
+        processor = BatchSpanProcessor(span_exporter=exporter, max_export_batch_size=1)
+        trace.set_tracer_provider(TracerProvider(resource=resource, active_span_processor=processor))
+        self.tracer = trace.get_tracer("system-tests-runner")
+
+    def test_main(self):
+        response = interfaces.backend._wait_for_trace(
```

Comment:
I suggest adding a helper method similar to the `assert_spans_exist()` or `assert_library_traces_exist()` that is going to be OTel specific to abstract away the specifics for the backend interaction. These are very simple wrappers around the `_wait_for_trace` function.
For example something like `backend.assert_otel_trace_exists(...)` should suffice.

If you really want the `_wait_for_trace` directly, we can make it non-private, and expose it properly, since now it's just an internal implementation of the other public methods.